sled-agent/config-reconciler/src/reconciler_task.rs

```diff
@@ -161,6 +161,12 @@ impl ReconcilerResult {
             self.latest_result.as_ref().map(|r| r.to_inventory());
         (status, latest_result)
     }
+
+    pub(crate) fn all_mounted_local_storage_datasets(
+        &self,
+    ) -> impl Iterator<Item = PathInPool> + '_ {
+        self.all_mounted_datasets_of_kind(DatasetKind::LocalStorage)
```

Comment:
Ug, this is going to walk right into a landmine I left, after _just_ talking about how I don't like to do this: https://github.com/oxidecomputer/omicron/blob/8a4860bc3dbc6af671b74ceeb50cf7df784fe1ae/sled-agent/config-reconciler/src/reconciler_task.rs#L242-L248

Can we either remove this method in favor of something like the `validate_local_storage_dataset_mounted()` proposed in an earlier comment, or fix the `match` to list all the `DatasetKind` variants explicitly and not panic on `DatasetKind::LocalStorage`?

Reply:
I fixed the match to list all variants explicitly in 7e3d4843b - I first tried moving `DatasetName`'s `full_name` routine to `DatasetKind` (to replace the mountpoint variant) but I think this should happen in another PR.